### ISSUE TITLE
Appsody deploy cmd can't detect exposed service URL in output logs

### ISSUE THREAD

**Issue by brightzheng100** [appsody/appsody#965]
**Describe the bug**

While using `ClusterIP` as service type:

```yaml
  service:
    annotations:
      prometheus.io/path: /actuator/prometheus
      prometheus.io/scrape: "true"
    port: 8080
    type: ClusterIP
```

The `appsody deploy` output logs would be wrong:

```sh
$ appsody deploy --no-build --namespace default
...
Attempting to get resource from Kubernetes ...
Running command: kubectl get route spring-boot-demo -o "jsonpath={.status.ingress[0].host}" --namespace default
Attempting to get resource from Kubernetes ...
Running command: kubectl get svc spring-boot-demo -o "jsonpath=http://{.status.loadBalancer.ingress[0].hostname}:{.spec.ports[0].nodePort}" --namespace default
Deployed project running at http://:
```

Please see the last line: `Deployed project running at http://:`

**Expected behavior**

Right detection should be applied with actionable advice to access the app deployed.

**If applicable please specify:**
 - CLI version: appsody 0.5.9
 - Stack you are using: springboot

**Comment by groeges** [comment]
Think this is a duplicate of https://github.com/appsody/appsody/issues/764

**Comment by neeraj-laad** [comment]
@groeges  @brightzheng100 if you both agree this is a duplicate, we can close this. 

@brightzheng100 we hope to have the fix in the next cli release.

**Comment by groeges** [comment]
@brightzheng100 What Kubernetes system are you using, ie local with docker desktop, OpenShift, etc?
What was the actual URL that you expected to get listed once the app was deployed?
Can you attach the output for the following commands to this issue:
```
kubectl get node -o yaml
``` 
```
kubectl get svc <your app name> -o yaml
```

**Comment by nastacio** [comment]
@neeraj-laad and @groeges , this is indeed similar to #764, but the symptoms seem slightly different.

In #764 the CLI still shows the port number, but in this case it does not show the IP address either.

**Comment by nastacio** [comment]
@groeges rightfully pointed out that `ClusterIP` really means no public IP address, so in my opinion the CLI output should either be:

1. A message indicating exactly that: [your deployment file specified ClusterIP, so it will only be accessible to other applications in the same cluster]
2. Same as "1", but also printing the internal host:ip, which may not be all that useful. It is possible that a K8s ingress or Istio ingress reference that IP, but realistically they will reference the application by hostname rather than IP address.

In short, I think "1" would be helpful to end-users. @brightzheng100 , what do you think?

**Comment by brightzheng100** [comment]
Thanks for the prompt responses @groeges @neeraj-laad @nastacio.

I think it shouldn't be any reason because of what kind of K8s I'm using -- well, I found and reported this case by using `Kind`.
This is really about showing some actionable result/advice for the user indicating how to access the deployed app and I believe the detection logic might not be 100% perfect to capture it -- from what the logs indicate, it's likely to look for the `{.status.loadBalancer.ingress[0].hostname}` which doesn't exist at least in my case.

I'd agree with @nastacio to detect the service type the app uses and show reasonable advice to the user. For example:
1. If `ClusterIP` is used, we may suggest to access it by using `kubectl port-forward`, or exposing it further by `oc expose` in OCP;
2. If `NodePort`, `LoadBalancer` etc. are used, we may show directly how to access it, or at least tell the user what to do next -- sometimes the LoadBalancer will take some time to provision the public IP so keeping the user to wait for the IP may not be a good idea.

What do you think?

**Comment by groeges** [comment]
@brightzheng100 I have a proposed change for when `ClusterIP` is specified in that the code will detect the ClusterIP:Port and then output something like this:
```
Deployed project running at http://172.21.62.246:9080
However as the ServiceType was specified as ClusterIP this url is only accessible to other applications in the same cluster.
To access it try using 'kubectl port-forward', or exposing it further by 'oc expose'
```
If `NodePort` is specified then we will output as we do today, but fixing the issue as mentioned in  https://github.com/appsody/appsody/issues/764

Still need to look at what gets output if `LoadBalancer` is specified. 

Does this seem reasonable and fit your requirement?
